pkg/kots/rego/kots-spec-opa-nonrendered.rego

```diff
@@ -951,11 +1074,51 @@ lint[output] {
   field := concat(".", [config_option.field, "when"])
 
   output := {
-    "rule": "config-option-when-is-invalid",
-    "type": "error",
+    "rule": rule_name,
+    "type": rule_config.level,
     "message": "Invalid \"when\" expression",
     "path": config_file_path,
     "field": field,
     "docIndex": config_data.docIndex
   }
 }
+
+# Check if LintConfig spec exists
+lintconfig_spec_exists {
+  file := files[_]
+  file.content.kind == "LintConfig"
+  file.content.apiVersion == "kots.io/v1beta1"
+}
+
+lint[output] {
+  rule_name := "missing-lintconfig-spec"
+  rule_config := lint_rule_config(rule_name, "warn")
+  not rule_config.off
+  not lintconfig_spec_exists
+  output := {
+    "rule": rule_name,
+    "type": rule_config.level,
+    "message": "Missing LintConfig spec"
+  }
+}
```

Comment:
since lint config is optional, Does this check needed?

Reply:
@pavansokkenagaraj  The level of each rule will now be read from the lint-config.yaml, so we need to have that file present if we want to show warnings/errors for the rules. Having the [default lint-config](https://github.com/replicatedhq/kots-default-yaml/pull/24/files) should make sure it's already present and the warning will show only if they delete it.